code/datums/ai/monkey/monkey_controller.dm

```diff
@@ -0,0 +1,211 @@
+/*
+AI controllers are a datumized form of AI that simulates the input a player would otherwise give to a mob. What this means is that these datums
+have ways of interacting with a specific mob and control it.
+*/
+///OOK OOK OOK
+
+/datum/ai_controller/monkey
+	blackboard = list(BB_MONKEY_AGRESSIVE = FALSE,\
+	BB_MONKEY_BEST_FORCE_FOUND = 0,\
+	BB_MONKEY_ENEMIES = list(),\
+	BB_MONKEY_BLACKLISTITEMS = list(),\
+	BB_MONKEY_PICKUPTARGET = null,\
+	BB_MONKEY_PICKPOCKETING = FALSE,
+	BB_MONKEY_DISPOSING = FALSE,
+	BB_MONKEY_TARGET_DISPOSAL = null,
+	BB_MONKEY_CURRENT_ATTACK_TARGET = null,
+	BB_MONKEY_CURRENT_ATTACK_TARGET)
+
+/datum/ai_controller/monkey/angry
+
+/datum/ai_controller/monkey/angry/TryPossessPawn(atom/new_pawn)
+	. = ..()
+	if(. & AI_CONTROLLER_INCOMPATIBLE)
+		return
+	blackboard[BB_MONKEY_AGRESSIVE] = TRUE //Angry cunt
+
+/datum/ai_controller/monkey/TryPossessPawn(atom/new_pawn)
+	if(!isliving(new_pawn))
+		return AI_CONTROLLER_INCOMPATIBLE
+	RegisterSignal(new_pawn, COMSIG_PARENT_ATTACKBY, .proc/on_attackby)
```

Comment:
I think you can register list of signals, don't quote me on that though

Reply:
I have never seen list being used in registering signals, only to unregister them. Also wouldn't that make it even less readable?